src/main/java/com/getindata/connectors/http/internal/SinkHttpClientResponse.java

```diff
@@ -11,21 +12,36 @@
 
 /**
  * Data class holding {@link HttpSinkRequestEntry} instances that {@link SinkHttpClient} attempted
- * to write, divided into two lists &mdash; successful and failed ones.
+ * to write.
  */
 @Data
 @ToString
 public class SinkHttpClientResponse {
 
     /**
-     * A list of successfully written requests.
+     * A list of requests along with write status.
      */
     @NonNull
-    private final List<HttpRequest> successfulRequests;
+    private final List<ResponseItem> requests;
 
-    /**
-     * A list of requests that {@link SinkHttpClient} failed to write.
-     */
-    @NonNull
-    private final List<HttpRequest> failedRequests;
+    public List<HttpRequest> getSuccessfulRequests() {
+        return requests.stream()
+                .filter(ResponseItem::isSuccessful)
+                .map(ResponseItem::getRequest)
+                .collect(Collectors.toList());
+    }
+
+    public List<HttpRequest> getFailedRequests() {
```

Comment:
I did a google and found. [https://stackoverflow.com/questions/47680711/which-http-errors-should-never-trigger-an-automatic-retry](https://stackoverflow.com/questions/47680711/which-http-errors-should-never-trigger-an-automatic-retry ) . It seems to be that you should only retry if the status code is retriable.  Maybe group into successful, retriable and failed (no retry).

Reply:
Thanks! Good point. I may refactor the code so that there are 3 statuses as you suggested. The change may also affect lookup, not only sink. So as a "side-effect" I may introduce lookup retry feature :)